components/brave_wallet/browser/internal/hd_key_sr25519.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_SR25519_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_SR25519_H_
+
+#include <memory>
+
+#include "base/containers/span.h"
+#include "base/types/pass_key.h"
+#include "brave/components/brave_wallet/browser/internal/polkadot/rust/sr25519.h"
+
+namespace brave_wallet {
+
+class HDKeySr25519 {
```

Comment:
I'm a bit curious about the fact that we have `HDKeySr25519` and `SchnorrkelKeyPair`, and one is just a thin layer over the other. I may be missing something, but it doesn't look like we are using `HDKeySr25519` to gatekeep access to anything important in `SchnorrkelKeyPair`.

I have discussed this a bit with @bridiver and we agreed that if this type is used only in wallet, and not in the browser code, and if the _public interface_ of `SchnorrkelKeyPair` is not exposing any cxxbridge types, then it would be better to merge `HDKeySr25519` and `SchnorrkelKeyPair` into a single type for simplicity.

Reply:
Actually, nevermind that question for this particular case. The only reason we were even using the passkey idiom was so that we could constructing using `std::make_unique` which isn't even required anymore, so there's no reason to even keep passkeys or friends around anymore. Everything can be much more directly coded.